p2p/protocol/identify/id.go

```diff
@@ -316,6 +344,18 @@ func (ids *IDService) populateMessage(mes *pb.Identify, c network.Conn) {
 	}
 	log.Debugf("%s sent listen addrs to %s: %s", c.LocalPeer(), c.RemotePeer(), laddrs)
 
+	// Generate a signed peer record containing our listen addresses and
+	// send it along with the unsigned addrs
+	signedRecord := ids.peerRecordManager.LatestRecord()
+	if ids.useSignedAddrs && signedRecord != nil {
```

Comment:
We'll need separate protocol versions for identify and identify push. This solution duplicates our addresses, in unsigned and signed forms; that could be made more efficient at the expense of 1 RTT to support legacy peers.

* identify:
  * `/ipfs/id/1.0.0`: deals with unsigned addresses.
  * `/p2p/id/1.1.0`: deals with signed addresses.

* identify push:
  * `/ipfs/id/push/1.0.0`: deals with unsigned addresses.
  * `/p2p/id/push/1.1.0`: deals with signed addresses.

When opening outbound streams, we'd negotiate 1.1.0 and 1.0.0, in that order. If settle on 1.1.0, we know the peer understands peer records and we can omit the unsigned counterparts; if we settle on 1.0.0, it would've cost us 1 extra RTT (which should disappear with ms2.0), but as more of the network upgrades, this cost will gradually decrease network-level.